src/components/mission/Missions.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import MissionItem from './MissionItem';
+import fetchMissionData from '../../apis/missionsApi';
+
+const Missions = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchMissionData());
+  }, []);
```

Comment:
As per task requirements, your component shouldn't be rendering data every time you switch routes.
Kindly, fix it.